crates/utils/src/lib.rs

```diff
@@ -45,24 +47,19 @@ pub const MAX_COMMENT_DEPTH_LIMIT: usize = 50;
 /// Doing DB transactions of bigger batches than this tend to cause seq scans.
 pub const DB_BATCH_SIZE: i64 = 1000;
 
+#[cfg(not(debug_assertions))]
 fn version() -> String {
-  if cfg!(debug_assertions) {
-    // For debug simply use the version from Cargo.toml. We can't use git_version here
-    // because it would cause a rebuild if any file in the repo is changed.
-    env!("CARGO_PKG_VERSION").to_string()
-  } else {
-    // Event cron means its a nightly build
-    // https://woodpecker-ci.org/docs/usage/environment
-    if option_env!("CI_PIPELINE_EVENT") == Some("cron") {
-      format!("nightly-{}", Utc::now().date_naive())
-    } else {
-      // For actual release builds use git binary for detailed version information.
-      git_version::git_version!(
-        args = ["--tags", "--dirty=-modified"],
-        fallback = env!("CARGO_PKG_VERSION")
-      )
-      .to_string()
-    }
+  match option_env!("LEMMY_VERSION") {
+    // For nightly add current date
+    Some("nightly") => format!("nightly-{}", chrono::Utc::now().date_naive()),
+    // Otherwise use the provided version name
+    Some(v) => v.to_string(),
+    // If no version was given take it from git
+    None => git_version::git_version!(
+      args = ["--tags", "--dirty=-modified"],
```

Comment:
This is the main issue. Remove this and move it to the docker build (and you'll run into the docker bug I commented on in the other issue). Its causing re-compiles on every git dirty modification.